File: solutions/typescript/VB_T3_007_list_contains.ts
// VB-T3-007: List Contains -- TypeScript baseline
type List = { tag: "Nil" } | { tag: "Cons"; head: number; tail: List };
function listContains(lst: List, target: number): boolean {
  switch (lst.tag) { case "Nil": return false; case "Cons": return lst.head === target || listContains(lst.tail, target); }
}
console.assert(listContains({ tag: "Cons", head: 1, tail: { tag: "Cons", head: 2, tail: { tag: "Nil" } } }, 2) === true);

File: solutions/python/VB_T3_007_list_contains.py
"""VB-T3-007: List Contains -- Python baseline."""
class List: pass
class Nil(List): pass
class Cons(List):
    __match_args__ = ("head", "tail")
    def __init__(self, head: int, tail: 'List'):
        self.head = head; self.tail = tail

def list_contains(lst: List, target: int) -> bool:
    match lst:
        case Nil(): return False
        case Cons(head, tail): return head == target or list_contains(tail, target)

if __name__ == "__main__":
    assert list_contains(Cons(1, Cons(2, Cons(3, Nil()))), 2) is True
    assert list_contains(Cons(1, Nil()), 5) is False
    print("VB-T3-007 ok")
